### PR TITLE
feat: allow setting of thousand/decimal separators

### LINKED ISSUE #636: Allow setting of thousand/decimal seperators
formatBalance & formatNumber

### PR DESCRIPTION
**Pull request by alko89**
Been trying to get an issue to get started with some development on polkadot and found this and gave it a go. 

Sets thousand and decimal separators based on locale.

Fix #636

### CODE REVIEW

File: packages/util/src/number/getSeparator.spec.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+// Copyright 2017-2023 @polkadot/util authors & contributors
```
Comment (jacogr) [inline packages/util/src/number/getSeparator.spec.ts:1]: I would keep this file in `format/*` and not export it from index. It really is specific to that functionality.

File: packages/util/src/number/getSeparator.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2017-2023 @polkadot/util authors & contributors
```
Comment (jacogr) [inline packages/util/src/number/getSeparator.ts:1]: As above with getSeperator.spec.ts - keep in `format/*`

File: packages/util/src/number/index.ts [hunk 3]
```diff
@@ -5,5 +5,6 @@
  * @summary Utility methods to convert to and from `number` values
  */
 
+export { getSeparator } from './getSeparator';
```
Comment (jacogr) [inline packages/util/src/number/index.ts:8]: Just remove the export after the previous moves.

**Review by jacogr — COMMENTED** [review]
Thank you. Think this is sensible and you did the right thing keeping the existing behavior and making the new opt-in. 

(My wife, which is German, is actually driven slightly batty by the `.` and `,` in the "wrong" positions on the apps UI, so having a setting there at some point to "default" (as-is) or "detected locale/specific locale" would at least make my life happier :))

**Comment by alko89** [comment]
Hehe, yeah we have separators inverted in Slovenia as well, I prefer to use en locale on my OS because I rather use the format that is more common on the internet. I'm annoyed by my bank website that forces sl locale on their site and have to be careful that I don't accidentally send someone 12.30 (1230 €) instead of 12,30 (12.3 €) :sweat_smile: 

Its annoying this is even a thing, but at least its not as big of a can of worms as...dates!

File: packages/util/src/number/getSeparator.spec.ts (hunk 1 above)
Comment (alko89) [reply packages/util/src/number/getSeparator.spec.ts]: Done, check

**Review by jacogr — APPROVED** [review]
Thank you!

**Pull request merged**